vulture/lines.py

```diff
@@ -23,27 +25,26 @@
     })
 
 
-# Reinventing wheel here to keep vulture lightweight.
-def memoize(func):
-    _cache = {}
+def _get_range(entry):
+    if isinstance(entry, ENTRY_POINTS):
+        last_body = entry.body[-1]
+        while isinstance(last_body, tuple(TRAVERSABLE_FIELDS.keys())):
+            fields = TRAVERSABLE_FIELDS[last_body.__class__]
+            for field in reversed(fields):
+                children = getattr(last_body, field)
+                if children:
+                    break
+            last_body = children[-1]  # The last child.
 
-    @wraps(func)
-    def wrapped(*args):
-        try:
-            return _cache[args]
-        except KeyError:
-            _cache[args] = result = func(*args)
-            return result
-    return wrapped
+        last_line = last_body.lineno
+        SourceRange = namedtuple("SourceRange",
+                                 ['name', 'start_line', 'last_line'])
+        source_range = SourceRange(entry.name, entry.lineno, last_line)
```

Comment:
Previously, I was passing all this around for tackling multi-line return statement. But then, I thought it would be better if we could get this up and running. 
It is indeed odd. 😛